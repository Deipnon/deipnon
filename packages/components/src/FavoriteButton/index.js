// @flow

import React from 'react'
import { IoIosHeartEmpty, IoIosHeart } from 'react-icons/io'
import styled from '@emotion/styled'
import { css } from '@emotion/core'

type PropsType = {|
    isSelected: boolean,
    onClick: (e: SyntheticEvent<HTMLSelectElement>) => void
|}

const EmptyIconStyle = css`
    color:grey;
`
const FiiledIconStyle = css`
    color:red;
`
const EmptyIcon = styled(IoIosHeartEmpty)(EmptyIconStyle)
const FiiledIcon = styled(IoIosHeart)(FiiledIconStyle)

const FavoriteButtonElement = (props: PropsType) => {
    if (props.isSelected) {
        return <FiiledIcon {...props}/>
    }
    return <EmptyIcon {...props}/>
}

const FavoriteButton = (props: PropsType) => <FavoriteButtonElement {...props}/>

export default FavoriteButton