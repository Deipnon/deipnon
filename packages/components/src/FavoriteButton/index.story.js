import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'

import FavoriteButton from '.'

const stories = storiesOf('FavoriteButton', module)
stories.addDecorator(withKnobs)

class FavoriteButtonSample extends React.Component {
    constructor (props) {
        super(props)

        this.state = {
            restorantId: '2',
            isSelected: false
        }
    }

    handleClickButton = () => {
        console.log('click me', this.state.restorantId)
        this.setState({
            isSelected: !this.state.isSelected
        })
    }

    render () {
        return (
            <FavoriteButton isSelected={this.state.isSelected} onClick={this.handleClickButton}/>
        )
    }
}

stories.add('default', () => <FavoriteButtonSample />)