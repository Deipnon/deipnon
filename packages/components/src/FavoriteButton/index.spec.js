import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import FavoriteButton from '.'

describe('FavoriteButton', () => {
  test('Empty', () => {
    const empty = renderer.create(<FavoriteButton isSelected='false'/>).toJSON()
    expect(empty).toMatchSnapshot()
  })
  test('Fiiled', () => {
    const fiiled = renderer.create(<FavoriteButton isSelected='true'/>).toJSON()
    expect(fiiled).toMatchSnapshot()
  })
})
