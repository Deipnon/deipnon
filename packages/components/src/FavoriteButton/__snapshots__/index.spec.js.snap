// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton Empty 1`] = `
<svg
  className="css-eomqaj"
  fill="currentColor"
  height="1em"
  isSelected="false"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 512 512"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M352 56h-1c-39.7 0-74.8 21-95 52-20.2-31-55.3-52-95-52h-1c-61.9.6-112 50.9-112 113 0 37 16.2 89.5 47.8 132.7C156 384 256 456 256 456s100-72 160.2-154.3C447.8 258.5 464 206 464 169c0-62.1-50.1-112.4-112-113z"
  />
</svg>
`;

exports[`FavoriteButton Fiiled 1`] = `
<svg
  className="css-eomqaj"
  fill="currentColor"
  height="1em"
  isSelected="true"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 512 512"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M352 56h-1c-39.7 0-74.8 21-95 52-20.2-31-55.3-52-95-52h-1c-61.9.6-112 50.9-112 113 0 37 16.2 89.5 47.8 132.7C156 384 256 456 256 456s100-72 160.2-154.3C447.8 258.5 464 206 464 169c0-62.1-50.1-112.4-112-113z"
  />
</svg>
`;
